Dashboard/Dashboard/Presentation/PrimaryCourseDashboardView.swift

```diff
@@ -183,6 +183,9 @@ public struct PrimaryCourseDashboardView<ProgramView: View>: View {
                     await viewModel.getEnrollments()
                 }
             }
+            .onAppear {
+                viewModel.updateNeeded = true
+            }
```

Comment:
This isn't working as expected. I guess the right way is to set this variable true on receiving the notification `onblockCompletionRequested` and then call `updateEnrollmentsIfNeeded` when view appears.

Reply:
The solution is working perfectly fine, but the issue is that it's a bit confusing. The app is launched and the enrollment API is just hit, and because of this part, the code is saying an update is still needed.

In ideal scenario, the `updateNeeded` only marked true when there is a need. Like if the completion call is called for a course that is not a primary course already.